server/src/main/java/org/elasticsearch/transport/ProxyConnectionStrategy.java

```diff
@@ -75,12 +72,12 @@
             Setting.Property.Dynamic, Setting.Property.NodeScope));
 
     /**
-     * Whether to include the hostname as a server_name attribute
+     * A configurable server_name attribute
      */
-    public static final Setting.AffixSetting<Boolean> INCLUDE_SERVER_NAME = Setting.affixKeySetting(
+    public static final Setting.AffixSetting<String> SERVER_NAME = Setting.affixKeySetting(
         "cluster.remote.",
-        "include_server_name",
-        (ns, key) -> boolSetting(key, false, new StrategyValidator<>(ns, key, ConnectionStrategy.PROXY),
+        "server_name",
```

Comment:
what's the upgrade path here?

Reply:
I'm not sure I understand the question here. Upgrade from a prior version? These settings are all new in 7.6.

Reply:
my bad, I did not see that this was in ProxyConnectionStrategy.